src/lib/utils.ts

```diff
@@ -170,7 +171,10 @@ export function encodeKeyValue(
  * @return The raw bytes key for the array element
  */
 export function encodeArrayKey(key: string, index: number) {
-  return key.slice(0, 34) + padLeft(numberToHex(index), 32).replace('0x', '');
+  return (
+    key.slice(0, 34) +
```

Comment:
Same here, suggest using the `slice(data, size)` function from viem library

Reply:
Could also use the `concat` method from viem here

<img width="753" alt="image" src="https://github.com/user-attachments/assets/20e27880-c0f8-43c6-9b86-2c7959da8552" />